rosetta-source/src/main/rosetta/product-qualification-func.rosetta

```diff
@@ -27,168 +57,154 @@ func Qualify_AssetClass_InterestRate: <"Qualifies a product as having the Asset
+                and economicTerms -> payout -> commodityPayout -> underlier exists)
                     // CO Basis
             or (economicTerms -> payout -> commodityPayout only exists
                 and economicTerms -> payout -> commodityPayout count = 2
-                and economicTerms -> payout -> commodityPayout -> underlier -> commodity count = 2)
+                and economicTerms -> payout -> commodityPayout -> underlier count = 2)
                     // CO Opt
             or (economicTerms -> payout -> optionPayout only exists
-                and if optionUnderlier exists
-                    then (Qualify_AssetClass_Commodity(
-                                    optionUnderlier -> contractualProduct -> economicTerms
-                                ) = True
-                            or Qualify_AssetClass_Commodity(
-                                    optionUnderlier -> security -> economicTerms
-                                ) = True
-                            or optionUnderlier -> commodity exists
-                            or optionUnderlier -> index ->> assetClass = AssetClassEnum -> Commodity
-                            or optionUnderlier -> security -> productTaxonomy -> primaryAssetClass any = AssetClassEnum -> Commodity)
-                    else False)
+                and (optionUnderlier -> Observable -> Asset -> Commodity exists
+                    or optionUnderlier -> Product -> TransferableProduct -> Commodity exists
+                    or if optionUnderlier -> Product exists
+                        then Qualify_AssetClass_Commodity(optionUnderlier -> Product ->> economicTerms) = True
+                        else False
+                ))          
                     // CO Fwd
-            or (((economicTerms -> payout -> forwardPayout, economicTerms -> payout -> fixedPricePayout) only exists
-                    or (economicTerms -> payout -> forwardPayout, economicTerms -> payout -> commodityPayout) only exists)
-                and (economicTerms -> payout -> forwardPayout -> underlier -> commodity exists
-                    or forwardUnderlier -> index ->> assetClass = AssetClassEnum -> Commodity
-                    or forwardUnderlier -> security -> productTaxonomy -> primaryAssetClass any = AssetClassEnum -> Commodity))
+            or (((economicTerms -> payout -> settlementPayout, economicTerms -> payout -> fixedPricePayout) only exists
+                    or (economicTerms -> payout -> settlementPayout, economicTerms -> payout -> commodityPayout) only exists)
+                and (economicTerms -> payout -> settlementPayout -> underlier -> Observable -> Asset -> Commodity exists
+                    or settlementUnderlier -> Product -> TransferableProduct -> Commodity exists
+                    or if settlementUnderlier -> Product ->> economicTerms exists
+                        then Qualify_AssetClass_Commodity(
+									settlementUnderlier -> Product ->> economicTerms) = True
```

Comment:
Same comment, is that qualification based on the settlement payout underlier's economic terms meant to cover the ETD case?